### PR TITLE
Edb stackup refactoring finalizing

### PR DESCRIPTION
**Pull request by maxcapodi78**
Removed all core_stackup methods and replaced with new stackup class accessible from Edb.stackup.
All legacy methods are there but pointing the new methods and added a deprecation warning

### CODE REVIEW

**Review by ring630 — COMMENTED** [review]
LGTM

**Review by ring630 — APPROVED** [review]

File: pyaedt/edb_core/stackup.py [hunk 1]
```diff
@@ -1837,128 +1640,636 @@ def place_a3dcomp_3d_placement(self, a3dcomp_path, angle=0.0, offset_x=0.0, offs
...
+        """
+        warnings.warn("Use `Edb.materials.get_layout_thickness` function instead.", DeprecationWarning)
+
+        return self._pedb.stackup.get_layout_thickness()
+
+    @pyaedt_function_handler()
+    def duplicate_material(self, material_name, new_material_name):
+        """Duplicate a material from the database.
+        It duplicates these five properties: ``permittivity``, ``permeability``, ` conductivity,``
+        ``dielectriclosstangent``, and ``magneticlosstangent``.
```
Comment (MaxJPRey) [inline pyaedt/edb_core/stackup.py:1962]:
```suggestion
        It duplicates these five properties: ``permittivity``, ``permeability``, `` conductivity``,
        ``dielectriclosstangent``, and ``magneticlosstangent``.
```

File: pyaedt/edb_core/materials.py [hunk 2]
```diff
@@ -654,3 +657,84 @@ def _load_materials(self, material=None):
...
+            original_material = self._edb.Definition.MaterialDef.FindByName(self._pedb._db, material_name)
+            if is_ironpython:  # pragma: no cover
+                property_box = _clr.StrongBox[float]()
+                original_material.GetProperty(self.material_name_to_id(property_name), property_box)
+                return float(property_box)
+            else:
+                _, property_box = original_material.GetProperty(
+                    self.material_name_to_id(property_name), self._edb_value(0.0)
+                )
+                return property_box.ToDouble()
```
Comment (MaxJPRey) [inline pyaedt/edb_core/materials.py:739]: Could a float be enough here?
```suggestion
                return float(property_box)
```

**Review by MaxJPRey — APPROVED** [review]
LGTM.

**Pull request merged**